src/compiler/factory/nodeFactory.ts

```diff
@@ -7517,11 +7515,3 @@ function mergeEmitNode(sourceEmitNode: EmitNode, destEmitNode: EmitNode | undefi
 
     return destEmitNode;
 }
-
-function mergeTokenSourceMapRanges(sourceRanges: (TextRange | undefined)[], destRanges: (TextRange | undefined)[]) {
-    if (!destRanges) destRanges = [];
-    for (const key in sourceRanges) {
-        destRanges[key] = sourceRanges[key];
-    }
-    return destRanges;
-}

```

Comment:
I read this function and it sort of scared me; what if the test has more elements than the source? Then I replaced it and nothing changed, so I kept going...

Reply:
`sourceRanges` and `destRanges` are holey arrays, so `length` doesn't actually matter. It probably should just be a `Map<number, TextRange>`, to be honest.